src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -462,6 +461,12 @@ def _exec_batch_with_process_pool(
 
         result_list = []
 
+        # Just for runtime delete the legacy related code
+        # Will removed later.
+        if self._flow_file is None:
```

Comment:
Should raise UnexpectedError with proper message.

Reply:
fixed